src/net/network_server.cpp

```diff
@@ -74,10 +74,11 @@ void NetworkServer::start_accept() {
 
     acceptor.async_accept(new_connection->get_socket(),
         [&, new_connection](const boost::system::error_code& error) {
+            int next_id = next_available_id();
```

Comment:
Why generate an ID if there's an error? Might as well move this into the if block.